oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -0,0 +1,189 @@
+        self.dir_fasta = os.path.join(self.dir_specificity, "fasta")
+        Path(self.dir_fasta).mkdir(parents=True, exist_ok=True)
+
+    def apply(self, oligo_DB, file_reference_DB, n_jobs):
+        """Apply the blastn filter in parallel on the given ``oligo_DB``. Each jobs filters a single gene, and  at the same time are generated at most ``n_job`` jobs.
+        The filtered database is returned.
+
+        :param oligo_DB: database containing the probes and their features
+        :type oligo_DB: dict
+        :param file_reference_DB: path to the file that will be used as reference for the alignement
+        :type file_reference_DB: str
+        :param n_jobs: number of simultaneous parallel computations
+        :type n_jobs: int
+        :return: probe info of user-specified genes
+        :rtype : dict
+        """
+
+        # create blast database
+        database_exists = False
+        database_name = os.path.basename(file_reference_DB)
+        # Check if blast database exists
+        for file in os.listdir(self.dir_blast):
+            if re.search(f"^{database_name}.*", file):
+                break
+
+        if not database_exists:
+            cmd = NcbimakeblastdbCommandline(
+                input_file=file_reference_DB,
+                dbtype="nucl",
+                out=os.path.join(self.dir_blast, database_name),
+            )
+            out, err = cmd()
+
+        # run the balst search
+        genes = list(oligo_DB.keys())
+        filtered_oligo_DBs = Parallel(n_jobs=n_jobs)(
+            delayed(self._run_blast)(oligo_DB[gene], gene, database_name)
+            for gene in genes
+        )
+
+        # reconstruct the oligos db and return it
+        for gene, filtered_oligo_DB in zip(genes, filtered_oligo_DBs):
+            oligo_DB[gene] = filtered_oligo_DB
+
+        return oligo_DB
+
+    def _run_blast(self, gene_DB, gene, database_name):
+        """Run BlastN alignment tool to find regions of local similarity between sequences, where sequences are probes and background sequences (e.g. transcript, genome, etc.).
+        BlastN identifies the transcript regions where probes match with a certain coverage and similarity.
+
+        :param gene_DB: database containing the probes form one gene
+        :type gene_DB: dict
+        :param gene: id of thet gene processed
+        :type gene: str
+        """
+        # run the blast search and write the results
+        file_probe_fasta_gene = self._create_fasta_file(gene_DB, self.dir_fasta, gene)
+        file_blast_gene = os.path.join(self.dir_blast, f"blast_{gene}.txt")
+        cmd = NcbiblastnCommandline(
+            query=file_probe_fasta_gene,
+            db=os.path.join(self.dir_blast, database_name),
+            outfmt="10 qseqid sseqid length qstart qend qlen",
+            out=file_blast_gene,
+            strand="plus",
+            word_size=self.word_size,
+            perc_identity=self.percent_identity,
+            num_threads=1,  # ????
```

Comment:
Good point! Need to do some benchmarking here. Last time, I tested with multiprocessing (tried several combination of `n_jobs`/ `n_threads` for the same total number of jobs). My conclusion was (oppositely to what I read everywhere) that it was faster to increase the number of jobs with `num_threads = 1`. To be confirmed with joblib